subprojects/docs/src/snippets/configurationCache/testKit/kotlin/src/test/kotlin/org/example/BuildLogicFunctionalTest.kt

```diff
@@ -0,0 +1,53 @@
+package org.example
+
+import org.gradle.testkit.runner.GradleRunner
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+import java.io.File
+
+class BuildLogicFunctionalTest {
+
+    @Rule
+    @JvmField
+    val testProjectDir: TemporaryFolder = TemporaryFolder()
+
+    lateinit var buildFile: File
+
+    @Before
+    fun setup() {
+        testProjectDir.newFile("settings.gradle").writeText("")
+        buildFile = testProjectDir.newFile("build.gradle")
+    }
+
+    // tag::functional-test-configuration-cache[]
+    @Test
+    fun `my task can be loaded from the configuration cache`() {
+
+        buildFile.writeText("""
+            plugins {
+                id 'org.example.my-plugin'
+            }
+        """)
+
+        runner()
+            .withArguments("--configuration-cache=on", "myTask")        // <1>
```

Comment:
⭕ If `--configuration-cache` is enough, I'd rather use it.

Reply:
We don't yet, see #13170